### PR TITLE
:sparkles: Validation for command-line flags

### PR DESCRIPTION
**Pull request by naveensrinivasan**
#### What kind of change does this PR introduce?

feature

(Is it a bug fix, feature, docs update, something else?)

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**
- Included validation for command line flags.
- [x] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
  https://github.com/ossf/scorecard/issues/1693

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

Included validation for command line flags.

```

### CODE REVIEW

File: options/options.go [hunk 1]
```diff
@@ -240,3 +247,21 @@ func validateFormat(format string) bool {
 		return false
 	}
 }
+
+// isCheckValidForLocalRepo returns an error if the check is not valid for local repository.
+func (o *Options) isCheckValidForLocalRepo() error {
```
Comment (azeemshaikh38) [inline options/options.go:252]: Each check self-declares a supported request type - https://github.com/ossf/scorecard/blob/main/checks/license.go#L39. Can use this to find whether the check is supported for Local or not. Here is an example - https://github.com/ossf/scorecard/blob/main/policy/policy.go#L211

File: options/options.go [hunk 2]
```diff
@@ -240,3 +247,21 @@ func validateFormat(format string) bool {
...
+	// This is a remote repository
+	if o.Local == "" {
+		return nil
+	}
```
Comment (azeemshaikh38) [inline options/options.go:256]: Use this `if` on the caller side instead? Something like:

```
if o.Local != "" {
  if err := os.isValidForLocalRepo(); err != nil {
    ...
  }
}
```

File: options/options.go (hunk 1 above)
Comment (azeemshaikh38) [inline options/options.go:252]: Nit: maybe name the fn `isValidForLocalRepo` since it is validating the `Options` struct and not a check.

**Review by azeemshaikh38 — COMMENTED** [review]
LGTM. Left some comments.

File: options/options.go (hunk 1 above)
Comment (naveensrinivasan) [reply options/options.go:258]: Took care of it. Thanks

Comment (naveensrinivasan) [reply options/options.go:258]: Yep, agreed. Took care of it.

File: options/options.go (hunk 2 above)
Comment (naveensrinivasan) [reply options/options.go:261]: I don't think this is applicable any more.

File: options/options.go [hunk 3]
```diff
@@ -240,3 +250,44 @@ func validateFormat(format string) bool {
...
+				"check %s is not supported for local repositories",
+				check,
+			)
+		}
+	}
+	return nil
+}
+
+// isValidForNonLocalRepo returns an error if the check is not valid for non-local repository.
+func (o *Options) isValidForNonLocalRepo() error {
```
Comment (laurentsimon) [inline options/options.go:276]: Maybe rename to `isValidForNonLocalGitRepo`. I initially thought this meant remote repos for checks that require GitHub APIs

**Review by laurentsimon — APPROVED** [review]
Thanks!

**Comment by azeemshaikh38** [comment]
@naveensrinivasan in our last sync you mentioned this PR needs more work. Let us know when this is ready for review.

**Comment by naveensrinivasan** [comment]
> @naveensrinivasan in our last sync you mentioned this PR needs more work. Let us know when this is ready for review.

It is ready fo review.

File: options/options.go (hunk 2 above)
Comment (azeemshaikh38) [reply options/options.go:258]: Resolving this, can use Laurent's suggestion.

File: options/options.go [hunk 4]
```diff
@@ -240,3 +247,21 @@ func validateFormat(format string) bool {
 		return false
 	}
 }
+
+// isCheckValidForLocalRepo returns an error if the check is not valid for local repository.
+func (o *Options) isCheckValidForLocalRepo() error {
+	// This is a remote repository
+	if o.Local == "" {
```
[the remark this answers is not in the document]
Comment (azeemshaikh38) [reply options/options.go:256]: +1

**Review by azeemshaikh38 — APPROVED** [review]

**Comment by naveensrinivasan** [comment]
Not stale.

**Pull request closed without merging**